### ISSUE TITLE
Right-click for context menu on LinkView causes vertex to be added

### ISSUE THREAD

**Issue by jnbarchan** [clientIO/joint#588]
I want the default behaviour that left-click on LinkView causes a vertex to be added at that point.  I _also_ am using context menu events on LinkView:

    linkView.on('cell:contextmenu', linkView.onCellContextMenu);

When user _right_-clicks on LinkView, JointJS creates a vertex before calling the context menu.  Clearly there is a need for JointJS to do something about _not_ creating a vertex on right-click, at least if code is handling context menu events.

**Comment by jnbarchan** [comment]
My workaround:

I need this addressed now because it is unusable as it is at present.  I do not want to alter JointJS code, though I have looked through its code to see what I can do given its present behaviour.

The best I have come up with is to _dynamically_ alter the return value of `Paper.options.interactive()` when user clicks on a `LinkView` depending on which button is being clicked:

1. Define a `Paper.options.interactive(cellView)` function, to control interaction setting, delegating to a `LinkView.interactive()` handler when user clicks on a `LinkView`.  If that returns `false`, interaction is prohibited, i.e. a vertex will not be created on the `LinkView`.
2. Handle the (new, undocumented? but useful) `link:pointerdown` event, which is passed the JavaScript `event` object, to store what button is being clicked.  I am storing this as a property on the `LinkView` object itself, so that...
3. ...Define a `LinkView.interactive()` handler which examines that property and returns `false` if it was not a left-click.

_If anyone wishes to suggest a better/less ugly workaround I would welcome it!_

This complexity is why I believe JointJS should handle _right-click-context-menu-instead-of-add-vertex_ itself, where it would be so much easier.

**Outline of Code:**
```
joint.dia.LinkView.extend({
  interactive: function ()
  {
    // return whether allowed to "interact" (perform a certain action) on progression
    // Paper.options.interactive() is set to call this when cell is a LinkView
    // see onLinkPointerDown() for explanation

    // if onLinkPointerDown() has been called and set that the pointerdown button was middle/right
    // forbid the action
    if (this.linkPointerDownButton === 1 || this.linkPointerDownButton === 2)
      return false;

    return true;
  },

  // Event Handlers
  onLinkPointerDown: function (evt, x, y)
  {
    // Event: pointer down on progression

    // Yuck:
    // JointJS performs actions on links (vertexAdd, vertexMove, vertexRemove, arrowheadMove)
    // on pointerdown regardless of which button is pressed
    // This means that when user right-clicks for context menu
    // he almost always gets a vertex created at point, prior to getting so far as to call onCellContextMenu() :(
    // To prevent that, we set "link:pointerdown" (which is an early event on a link) to call this function
    // Here we have to note what button was pressed on the LinkView
    // Then, when LinkView.interactive() is called later on in the processing (via Paper.options.interactive())
    // it can look at this and decide whether to allow/block the link view action

    this.linkPointerDownButton = evt.button;
  },

  onCellContextMenu: function (evt, x, y)
  {
    // ....
  }
});

joint.dia.Paper.extend({
  interactive: function (cellView)
  {
    // return whether cellView is allowed to "interact" (perform a certain action)

    // for a LinkView call its interactive() function to decide
    // see LinkView .interactive() & LinkView .onLinkPointerDown()
    if (cellView instanceof joint.dia.LinkView &&
        typeof(cellView.interactive) === 'function')
      return cellView.interactive();

    // else always allow interaction
    return true;
  }
});

paper.options.interactive = paper.interactive;
linkView.on('link:pointerdown', linkView .onLinkPointerDown);
linkView.on('cell:contextmenu', linkView .onCellContextMenu);
```

**Comment by kumilingus** [comment]
Fixed in `v2.1.3`.
